compiler/rustc_type_ir/src/interner.rs

```diff
@@ -106,6 +106,7 @@ pub trait Interner:
     type ParamTy: ParamLike;
     type BoundTy: BoundVarLike<Self>;
     type PlaceholderTy: PlaceholderLike<Self, Bound = Self::BoundTy>;
+    type Symbol: Copy + Hash + PartialEq + Eq + TypeFoldable<Self> + TypeVisitable<Self>;
```

Comment:
```suggestion
    type Symbol: Copy + Hash + PartialEq + Eq;
```
use `#[type_foldable(identity)]` and #[type_visitable(skip)]` for all uses of `Symbol` instead